tests/type_propagation/gold/null_equals_test.gold

```diff
@@ -70,7 +70,7 @@ A tests/type_propagation/null_equals_test.toit
 
 A.== tests/type_propagation/null_equals_test.toit
  - argument 0: {A}
- - argument 1: {String_|Null_|SmallInteger_|A}
+ - argument 1: {*}
```

Comment:
This got worse. I'll need to understand why before submitting this.

Reply:
I now understand what's happening. Inside a lambda method, we don't know what the type of 'this' is in the type propagation phase. When we inline the 'load field' bytecode that loads from 'this' into the lambda method -- instead of calling a named getter -- we get broader types, because stores to all fields with the same index (not the same name anymore) flow into it. 

This can be addressed by dealing better with captured variables (in particular final ones) for lambdas. All for another PR.